bind/gen.go

```diff
@@ -0,0 +1,666 @@
+		return true
+	}
+	return false
+}
+
+func complex64GoToPy(c complex64) *C.PyObject {
+	return C.PyComplex_FromDoubles(C.double(real(c)), C.double(imag(c)))
+}
+
+func complex64PyToGo(o *C.PyObject) complex64 {
+	v := C.PyComplex_AsCComplex(o)
+	return complex(float32(v.real), float32(v.imag))
+}
+
+func complex128GoToPy(c complex128) *C.PyObject {
+	return C.PyComplex_FromDoubles(C.double(real(c)), C.double(imag(c)))
+}
+
+func complex128PyToGo(o *C.PyObject) complex128 {
+	v := C.PyComplex_AsCComplex(o)
+	return complex(float64(v.real), float64(v.imag))
+}
+
+%[9]s
+`
+
+	goExePreambleC = `
+#if PY_VERSION_HEX >= 0x03000000
+extern PyObject* PyInit__%[1]s(void);
+static inline void gopy_load_mod() {
+	PyImport_AppendInittab("_%[1]s", PyInit__%[1]s);
+}
+#else
+extern void* init__%[1]s(void);
+static inline void gopy_load_mod() {
+	PyImport_AppendInittab("_%[1]s", init__%[1]s);
+}
+#endif
+
+`
+
+	goExePreambleGo = `
+// wchar version of startup args
+var wargs []*C.wchar_t
+
+//export GoPyMainRun
+func GoPyMainRun() {
+	// need to encode char* into wchar_t*
+	for i := range os.Args {
+		wargs = append(wargs, C.Py_DecodeLocale(C.CString(os.Args[i]), nil))
+	}
+	C.gopy_load_mod()
+	C.Py_Initialize()
+	C.PyEval_InitThreads()
+	C.Py_Main(C.int(len(wargs)), &wargs[0])
+}
+
+`
+
+	PyBuildPreamble = `# python build stubs for package %[1]s
+# File is generated by gopy. Do not edit.
+# %[2]s
+
+from pybindgen import retval, param, Module
+import sys
+
+mod = Module('_%[1]s')
+mod.add_include('"%[1]s_go.h"')
+mod.add_function('GoPyInit', None, [])
+`
+
+	// 3 = specific package name, 4 = spec pkg path, 5 = doc, 6 = imports
+	PyWrapPreamble = `%[5]s
+# python wrapper for package %[4]s within overall package %[1]s
+# This is what you import to use the package.
+# File is generated by gopy. Do not edit.
+# %[2]s
+
+# the following is required to enable dlopen to open the _go.so file
+import os,sys,inspect,collections
+cwd = os.getcwd()
+currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+os.chdir(currentdir)
+import _%[1]s
+os.chdir(cwd)
+
+# to use this code in your end-user python file, import it as follows:
+# from %[1]s import %[3]s
+# and then refer to everything using %[3]s. prefix
+# packages imported by this package listed below:
+
+%[6]s
+
+`
+
+	// exe version of preamble -- doesn't need complex code to load _ module
+	// 3 = specific package name, 4 = spec pkg path, 5 = doc, 6 = imports
+	PyWrapExePreamble = `%[5]s
+# python wrapper for package %[4]s within standalone executable package %[1]s
+# This is what you import to use the package.
+# File is generated by gopy. Do not edit.
+# %[2]s
+
+import _%[1]s, collections
+
+# to use this code in your end-user python file, import it as follows:
+# from %[1]s import %[3]s
+# and then refer to everything using %[3]s. prefix
+# packages imported by this package listed below:
+
+%[6]s
+
+`
+
+	GoPkgDefs = `
+import collections
+	
+class GoClass(object):
```

Comment:
couldn't this be just `Class`?
the name of the module being `go`...

Reply:
just not sure if python is as well-insulated namespace-wise compared to Go -- and lower-case "class" is a keyword so visually, it just seemed like GoClass might be safer?

Reply:
Any final thoughts about this?  I could go either way..